cartographer/mapping/sparse_pose_graph.h

```diff
@@ -87,10 +89,9 @@ class SparsePoseGraph {
   virtual void AddNodeFromProto(const transform::Rigid3d& global_pose,
                                 const proto::Node& node) = 0;
 
-  // Adds serialized constraints. The corresponding trajectory nodes and submaps
-  // have to be deserialized before calling this function.
-  virtual void AddSerializedConstraints(
-      const std::vector<Constraint>& constraints) = 0;
+  // Adds constraints and IMU data from a proto. The corresponding trajectory
+  // nodes and submaps have to be deserialized before calling this function.
+  virtual void AddDataFromProto(const proto::SparsePoseGraph& proto) = 0;
```

Comment:
This is not a great interface since it combines two orthogonal concerns: constraints and IMU data. `AddImuData()` already exists, so there is also no need to change `AddSerializedConstraints()`.

Reply:
My reasoning was the following: the sparse pose graph owns the optimization problem and the IMU data contained therein, so I thought that `SPG::ToProto` should be responsible for serializing that. Following that, since we would be serializing everything (constraints and imu data and later odometry) in `SPG::ToProto`, I thought that the single inverse function (resembling `FromProto`) should also handle everything, so I proposed the rename.

Another concern was that if I use `AddImuData` to load all IMU messages, this would be slower since there's a really big number of them and we would need to push a work item for each one.

I considered storing it into the proto stream instead, as you wrote in the review, but I thought that since we're already storing all node poses inside the SPG proto, IMU data wouldn't make much of a difference. But sure, if you would prefer to store it in the proto stream, we can do that as well.

Reply:
Yes, please update to store IMU data in the proto stream. We rather want to split the current `SparsePoseGraph` message into multiple messages, not add more data. There is also much more IMU data (e.g. 250 Hz in our demo bags) usually then poses. You might run into the proto message default limit for long demo bags already.